packages/radio-player/src/search-handler/search-index.ts

```diff
@@ -0,0 +1,139 @@
+   * individual entries. This allows searches to cross over transcript entries.
+   *
+   * NOTE: When the mergedTranscript gets created, spaces are put between each transcript
+   * entry, otherwise the words would run into each other. We account for this in the
+   * indices above.
+   *
+   * @type {string}
+   * @memberof SearchIndexCache
+   */
+  readonly mergedTranscript: string = '';
+
+  readonly mergedTranscriptLowercased: string = '';
+
+  /**
+   * This gets populated as part of the search index build. It maps the start and end indicies
+   * of all of the transcript entries so we can quickly look up where an entry is in the
+   * overall transcript.
+   *
+   * @type {TranscriptEntryRange[]}
+   * @memberof SearchIndexCache
+   */
+  readonly transcriptEntryRanges: TranscriptEntryRange[] = [];
+
+  constructor(mergedTranscript = '', transcriptEntryRanges: TranscriptEntryRange[] = []) {
+    this.mergedTranscript = mergedTranscript;
+    this.transcriptEntryRanges = transcriptEntryRanges;
+    this.mergedTranscriptLowercased = mergedTranscript.toLowerCase();
+  }
+}
+
+/* eslint-disable import/prefer-default-export */
+export class SearchIndex {
+  /**
+   * Make these properties readonly. We don't want outside classes touching the private storage.
+   *
+   * @readonly
+   * @type {string}
+   * @memberof SearchIndex
+   */
+  get mergedTranscript(): string {
+    return this.searchIndexCache.mergedTranscript;
+  }
+
+  get transcriptEntryRanges(): TranscriptEntryRange[] {
+    return this.searchIndexCache.transcriptEntryRanges;
+  }
+
+  /**
+   * This gets populated as part of the search index build. It merges all of the transcript
+   * entries together so we can search it as a single document instead of a bunch of
+   * individual entries. This allows searches to cross over transcript entries.
+   *
+   * It also indexes the entries so we can quickly look them up later.
+   *
+   * @private
+   * @memberof SearchIndex
+   */
+  private searchIndexCache: SearchIndexCache = new SearchIndexCache();
+
+  /**
+   * Find the closes TranscriptEntryRange to the given overallCharIndex
+   *
+   * @param overallCharIndex
+   * @returns {TranscriptEntryRange | undefined}
+   */
+  getTranscriptEntryAt(overallCharIndex: number): TranscriptEntryRange | undefined {
+    return this.searchIndexCache.transcriptEntryRanges.find(entry => {
+      const { range } = entry;
+      return range.endIndex > overallCharIndex && range.startIndex <= overallCharIndex;
+    });
+  }
+
+  /**
+   * Finds all of the ranges of all the search results across the entire transcript.
+   *
+   * @private
+   * @param {string} term
+   * @returns {Range[]}
+   * @memberof SearchIndex
+   */
+  getSearchRanges(term: string): Range[] {
+    const ranges: Range[] = [];
+    const termLowerCased: string = term.toLowerCase();
+
+    let index = -1;
+    /* eslint-disable-next-line no-cond-assign */
```

Comment:
instead of disabling, can we just push a `new Range` into ranges?

Reply:
Hmm maybe? I _think_ it was complaining about the `index = this.searchIndexCache.me...` in the `while` loop, not so much the `push(newRange)` part.

I'm not sure how to make this loop more readable, but definitely up for it.